pir/cpp/server.cpp

```diff
@@ -47,9 +48,85 @@ StatusOr<std::unique_ptr<PIRServer>> PIRServer::Create(
 
 StatusOr<PIRPayload> PIRServer::ProcessRequest(
     const PIRPayload& payload) const {
-  ASSIGN_OR_RETURN(auto result, db_->multiply(payload.Get()));
+  if (payload.Get().size() != 1) {
+    return InvalidArgumentError("Number of ciphertexts in request must be 1");
+  }
+  if (!payload.GetKeys()) {
+    return InvalidArgumentError("Must have Galois Keys in request");
+  }
+
+  auto selection_vector =
+      oblivious_expansion(payload.Get()[0], DBSize(), *payload.GetKeys());
+
+  ASSIGN_OR_RETURN(auto mult_results, db_->multiply(selection_vector));
+
+  seal::Ciphertext result;
+  context_->Evaluator()->add_many(mult_results, result);
```

Comment:
Not sure what you mean?

Reply:
cat add_many output a zero ciphertext? 
Can mult_results be ciphertexts over [1,1] and [-1, -1], they sum resulting in an encrypted zero? SEAL will throw exception in that scenario.

Reply:
Not sure where you got that from, but I've written a unit test for that and we don't get any kind of exception.

Reply:
This is the line I'm referring to. If it's impossible for us to get an transparent output, it's ok https://github.com/microsoft/SEAL/blob/master/native/src/seal/evaluator.cpp#L161

Reply:
Have you tried to do PIR on a database with elements 0 and 1? (One element per row, no packing.)

It might throw such an error (and even if not, it would be good to add as unit test).

Reply:
No, but I wrote an equivalent unit test that always produces zero output and it seems to pass